web/src/left_sidebar_navigation_area_popovers.js

```diff
@@ -27,6 +29,20 @@ function common_click_handlers() {
         channel.patch({
             url: "/json/settings",
             data,
+            success() {
+                if (default_view === settings_config.default_view_values.all_messages.code) {
+                    $(".recent-view-sidebar-menu-icon").removeClass("hide");
+                    $(".all-messages-sidebar-menu-icon").addClass("hide");
+                } else if (
+                    default_view === settings_config.default_view_values.recent_topics.code
+                ) {
+                    $(".recent-view-sidebar-menu-icon").addClass("hide");
+                    $(".all-messages-sidebar-menu-icon").removeClass("hide");
+                } else {
+                    $(".recent-view-sidebar-menu-icon").removeClass("hide");
+                    $(".all-messages-sidebar-menu-icon").removeClass("hide");
+                }
+            },
```

Comment:
This logic needs to go in a function called from `server_events_dispatch.js`, or we'll have stale state when a different browser window makes the request to change the default view.

Reply:
Oops! Done.